modules/runner/src/main/java/org/apache/ignite/internal/app/IgniteImpl.java

```diff
@@ -785,19 +786,23 @@ public CompletableFuture<Ignite> start(Path configPath) {
         ExecutorService startupExecutor = Executors.newSingleThreadExecutor(NamedThreadFactory.create(name, "start", LOG));
 
         try {
+            List<CompletableFuture<Void>> startFuturesBeforeJoin = new ArrayList<>();
```

Comment:
This looks suspicious. I would expect all components to start sequentially for now (i.e. by chaining the start futures), not in parallel. This should probably be encapsulated inside the `lifecycleManager`.

Reply:
Why does it look suspicious? Synchronous parts still start sequentially, as it were before.